backend/backend/settings/deps/oidc.py

```diff
@@ -0,0 +1,105 @@
+import logging
+import os
+
+import requests
+
+from .. import common
+from . import ledger
+
+_LOGGER = logging.getLogger(__name__)
+
+# mozilla-django-oidc values are OIDC_*
+# but to avoid polluting the global namespace
+# we'll use an "OIDC" dict for any setting added by us
+
+OIDC = {
+    "ENABLED": common.to_bool(os.environ.get("OIDC_ENABLED", "false")),
+    "USERS": {},
+    "OP": {},
+}
+if OIDC["ENABLED"]:  # noqa: C901
+    common.INSTALLED_APPS += ["mozilla_django_oidc"]  # load after auth
+    common.AUTHENTICATION_BACKENDS += ["users.authentication.OIDCAuthenticationBackend"]
+    common.LOGGING["loggers"]["mozilla_django_oidc"] = {
+        "level": common.LOG_LEVEL,
+        "handlers": ["console"],
+        "propagate": False,
+    }
+
+    OIDC["USERS"]["APPEND_DOMAIN"] = common.to_bool(os.environ.get("OIDC_USERS_APPEND_DOMAIN", "false"))
+
+    OIDC["USERS"]["DEFAULT_CHANNEL"] = os.environ.get("OIDC_USERS_DEFAULT_CHANNEL")
+    if not OIDC["USERS"]["DEFAULT_CHANNEL"]:
+        raise Exception("No default channel provided for OIDC users")
+    if OIDC["USERS"]["DEFAULT_CHANNEL"] not in ledger.LEDGER_CHANNELS:
+        raise Exception(f"Channel {OIDC['USERS']['DEFAULT_CHANNEL']} does not exist")
+    OIDC["USERS"]["LOGIN_VALIDITY_DURATION"] = int(
+        os.environ.get("OIDC_USERS_LOGIN_VALIDITY_DURATION", 60 * 60)  # seconds
+    )
+
+    OIDC_AUTHENTICATE_CLASS = "users.views.authentication.OIDCAuthenticationRequestView"
+    OIDC_CALLBACK_CLASS = "users.views.authentication.OIDCAuthenticationCallbackView"
+
+    OIDC_RP_CLIENT_ID = os.environ.get("OIDC_RP_CLIENT_ID")
+    OIDC_RP_CLIENT_SECRET = os.environ.get("OIDC_RP_CLIENT_SECRET")
+    OIDC_RP_SIGN_ALGO = os.environ.get("OIDC_RP_SIGN_ALGO")
+
+    OIDC_OP_AUTHORIZATION_ENDPOINT = os.environ.get("OIDC_OP_AUTHORIZATION_ENDPOINT")
+    OIDC_OP_TOKEN_ENDPOINT = os.environ.get("OIDC_OP_TOKEN_ENDPOINT")
+    OIDC_OP_USER_ENDPOINT = os.environ.get("OIDC_OP_USER_ENDPOINT")
+    OIDC_OP_JWKS_ENDPOINT = os.environ.get("OIDC_OP_JWKS_URI")
+
+    OIDC["OP"]["URL"] = os.environ.get("OIDC_OP_URL").removesuffix("/")
+    OIDC["OP"]["DISPLAY_NAME"] = os.environ.get("OIDC_OP_DISPLAY_NAME")
+    if not OIDC["OP"]["DISPLAY_NAME"]:
+        OIDC["OP"]["DISPLAY_NAME"] = OIDC["OP"]["URL"]
+
+    # put the OP URL before the endpoints, but not JWKS
+    # (it is specified as URI, not endpoint, despite the mozilla_django_oidc name)
+    for endpoint in ["OIDC_OP_AUTHORIZATION_ENDPOINT", "OIDC_OP_TOKEN_ENDPOINT", "OIDC_OP_USER_ENDPOINT"]:
+        if globals()[endpoint]:
+            globals()[endpoint] = OIDC["OP"]["URL"] + globals()[endpoint]
```

Comment:
The reason is it looks weird in the values. But it is better without (simpler and more flexible, and at not much cost since it's off the happy path anyway).